cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
+        return self._is_any_custom_api_name(api_name) and (
+            (
+                self.options["namespace_prefix"]
+                and api_name.startswith(self.options["namespace_prefix"])
+            )
+            or api_name.count("__") == 1
+        )
+
+    def _is_core_field(self, api_name):
+        # This list will be revised when we change our user support.
+        # We always include these fields on all objects (if present)
+        return api_name in self.core_fields
+
+    def _is_object_mappable(self, obj):
+        return not any(
+            [
+                obj["name"] in self.options["ignore"],  # User-specified exclusions
+                obj["name"].endswith(
+                    "ChangeEvent"
+                ),  # Change Data Capture entities (which get custom fields)
+                obj["name"].endswith("__mdt"),  # Custom Metadata Types (MDAPI only)
+                obj["name"].endswith("__e"),  # Platform Events
+                obj["customSetting"],  # Not Bulk API compatible
+                obj["name"]  # Objects we can't or shouldn't load/save
+                in [
+                    "User",
+                    "Group",
+                    "LookedUpFromActivity",
+                    "OpenActivity",
+                    "Task",
+                    "Event",
+                    "ActivityHistory",
+                ],
+            ]
+        )
+
+    def _is_field_mappable(self, obj, field):
+        return not any(
+            [
+                "{}.{}".format(obj, field["name"])  # User-ignored list
+                in self.options["ignore"],
+                "(Deprecated)" in field["label"],  # Deprecated managed fields
+                field["type"] == "base64",  # No Bulk API support for base64 blob fields
+                not field["createable"],  # Non-writeable fields
+                field["type"] == "reference"  # Self-lookups
+                and field["referenceTo"] == [obj],
+                field["type"] == "reference"  # Outside lookups
+                and not self._are_lookup_targets_in_operation(field),
+            ]
+        )
+
+    def _is_required_field(self, field):
+        return (field["createable"] and not field["nillable"]) or (
+            field["type"] == "reference" and field["relationshipOrder"] == 1
+        )
+
+    def _has_our_custom_fields(self, obj):
+        return any(
+            [self._is_our_custom_api_name(field["name"]) for field in obj["fields"]]
+        )
+
+    def _are_lookup_targets_in_operation(self, field):
+        return all([f in self.mapping_objects for f in field["referenceTo"]])
+
+    def _is_lookup_to_included_object(self, field):
+        return field["type"] == "reference" and self._are_lookup_targets_in_operation(
+            field
+        )
+
+    def _collect_objects(self):
+        self.mapping_objects = []
+
+        # Cache the global describe, which we'll walk.
+        self.global_describe = self.sf.describe()
```

Comment:
We should log something before starting this so it doesn't look like it's doing nothing.